fix: downgrade go-openapi packages to resolve build failure #12185

7908837174 opened the pull request:
- Downgraded github.com/go-openapi/validate from v0.24.0 to v0.23.0
- Downgraded github.com/go-openapi/runtime from v0.28.0 to v0.27.2
- Resolves build compatibility issue with prometheus/client_golang v1.20.2+

The go-openapi/validate v0.24.0 package had missing type definitions (SpecValidator, SchemaValidatorOptions, Result, etc.) that were causing undefined symbol errors during build. The downgrade to compatible versions restores functionality.

Fixes #12185

Tested:
- Server builds successfully with 'go build'
- All go-openapi dependencies are now compatible
- No breaking changes to existing functionality

**Notes for Reviewers**

- This PR fixes #

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits.
4. Include before and after screenshots/terminal output.

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

7908837174 commented:
Thank you for your support!

On Mon, 30 Jun, 2025, 17:29 gemini-code-assist[bot], <
***@***.***> wrote:

> ***@***.***[bot]* commented on this pull request.
> Code Review
>
> This pull request correctly downgrades go-openapi/runtime and
> go-openapi/validate to resolve a build failure. The changes in go.mod and
> go.sum are appropriate. I've added a couple of suggestions to include
> comments in go.mod to document the reason for these downgrades, which
> will improve long-term maintainability by preventing accidental upgrades
> that could re-introduce the issue. Otherwise, the changes look good.
> ------------------------------
>
> In go.mod
> <https://github.com/meshery/meshery/pull/15253#discussion_r2174906717>:
>
> > @@ -27,7 +27,7 @@ require (
>  	github.com/fatih/color v1.18.0
>  	github.com/fsnotify/fsnotify v1.9.0
>  	github.com/ghodss/yaml v1.0.0
> -	github.com/go-openapi/runtime v0.28.0
> +	github.com/go-openapi/runtime v0.27.2
>
> [image: medium]
> <https://camo.githubusercontent.com/a33c7ba47779a2aabda978d90fc2665dd949d88da87f5ff554e65308de059f02/68747470733a2f2f7777772e677374617469632e636f6d2f636f64657265766965776167656e742f6d656469756d2d7072696f726974792e737667>
>
> To improve long-term maintainability, it's helpful to add a comment
> explaining why this dependency is being downgraded. This context is
> valuable for future developers who might otherwise try to update it, and it
> prevents the reintroduction of the build issue. Referencing the issue
> number is a great way to do this.
>
> 	github.com/go-openapi/runtime v0.27.2 // See #12185
>
> ------------------------------
>
> In go.mod
> <https://github.com/meshery/meshery/pull/15253#discussion_r2174906719>:
>
> > @@ -185,7 +185,7 @@ require (
>  	github.com/go-openapi/loads v0.22.0 // indirect
>  	github.com/go-openapi/spec v0.21.0 // indirect
>  	github.com/go-openapi/swag v0.23.0 // indirect
> -	github.com/go-openapi/validate v0.24.0 // indirect
> +	github.com/go-openapi/validate v0.23.0 // indirect
>
> [image: medium]
> <https://camo.githubusercontent.com/a33c7ba47779a2aabda978d90fc2665dd949d88da87f5ff554e65308de059f02/68747470733a2f2f7777772e677374617469632e636f6d2f636f64657265766965776167656e742f6d656469756d2d7072696f726974792e737667>
>
> Similarly, please add a comment here to explain the version pin for this
> indirect dependency. This ensures that the reason for the downgrade is
> clear, even for transitive dependencies.
>
> 	github.com/go-openapi/validate v0.23.0 // indirect; See #12185
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/meshery/meshery/pull/15253#pullrequestreview-2970961111>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BLR2IIOOTJ7DVUP7KM26VLT3GERC5AVCNFSM6AAAAACAOALSNSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDSNZQHE3DCMJRGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

7908837174 commented:
Thanks for the feedback!

On Mon, 30 Jun, 2025, 17:58 github-actions[bot], ***@***.***>
wrote:

> *github-actions[bot]* left a comment (meshery/meshery#15253)
> <https://github.com/meshery/meshery/pull/15253#issuecomment-3018964702>
>
> *Commit SHA:* c901299aace2e2e5c9c6a21f2022915de9db78cd
> END-TO-END TESTS
>
>    - Testing started at: June 30th 2025, 12:20:45 pm
>
> *📦 Test Result Summary*
>
>    - ✅ 76 passed
>    - ❌ 0 failed
>    - ⚠️ 0 flaked
>    - ⏩ 0 skipped
>
> ⌛ *Duration: 5 minutes and 31 seconds*
>
> *Overall Result*: 👍 All tests passed.
> [Show/Hide] Test Result Details
> Test Browser Test Case Tags Result
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/meshery/meshery/pull/15253#issuecomment-3018964702>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BLR2IIKCMJGBNXTVC7UNWVT3GEUNHAVCNFSM6AAAAACAOALSNSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTAMJYHE3DINZQGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

7908837174 commented on go.mod:
```diff
@@ -27,7 +27,7 @@ require (
 	github.com/fsnotify/fsnotify v1.9.0
 	github.com/ghodss/yaml v1.0.0
-	github.com/go-openapi/runtime v0.28.0
+	github.com/go-openapi/runtime v0.27.2
```
Added explanatory comments to both downgraded packages in the [go.mod](https://potential-space-carnival-q7pvw6vx4w9rcp4v.github.dev/) file:

github.com/go-openapi/runtime v0.27.2 // Downgraded to maintain compatibility with validate v0.23.0 - See #12185
github.com/go-openapi/validate v0.23.0 // Downgraded due to build compatibility issues with prometheus client - See #12185
Committed the changes with a clear commit message explaining the purpose of these comments

Pushed the updated branch to the same PR branch.

Benefits of Adding These Comments:
Prevents accidental upgrades: Future developers will understand why these specific versions are locked
Provides context: References issue #12185 for full background information
Improves maintainability: Makes the codebase more self-documenting
Saves time: Prevents developers from spending time debugging the same issue again.

The branch fix/go-openapi-validate-compatibility now contains both the technical fix and the documentation to prevent regression. The PR is ready for review and will provide a complete solution to issue #12185.

l5io commented:
🚀 Preview for commit 2aa1f301443688a58cd5786f39961f997a786432 at: https://6864be1990e786792281b73b--meshery-docs-preview.netlify.app

7908837174 commented:
Thanks for the feedback!

On Wed, 2 Jul, 2025, 11:40 github-actions[bot], ***@***.***>
wrote:

> *github-actions[bot]* left a comment (meshery/meshery#15253)
> <https://github.com/meshery/meshery/pull/15253#issuecomment-3026577621>
>
> *Commit SHA:* 2a84919f74f5cef60a812481556913f7ce6053c8
> END-TO-END TESTS
>
>    - Testing started at: July 2nd 2025, 6:04:05 am
>
> *📦 Test Result Summary*
>
>    - ✅ 76 passed
>    - ❌ 0 failed
>    - ⚠️ 0 flaked
>    - ⏩ 0 skipped
>
> ⌛ *Duration: 5 minutes and 21 seconds*
>
> *Overall Result*: 👍 All tests passed.
> [Show/Hide] Test Result Details
> Test Browser Test Case Tags Result
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/meshery/meshery/pull/15253#issuecomment-3026577621>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BLR2IINT6SRYJHMSRCSTJBT3GNZT7AVCNFSM6AAAAACAOALSNSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTAMRWGU3TONRSGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

vishalvivekm commented:
@7908837174 
Thank you for your contributions! Let's discuss this today, during Meshery Development call at 9:00 AM CT. Add it as an agenda item to the 📝 [meeting minutes](https://meet.meshery.io/dev-minutes), if you would. https://meet.meshery.io/dev

7908837174 commented:
> too many conflicts, too many changes. Are these desired?

OK

7908837174 commented:
> @7908837174 Would you mind solving thr merge conflicts.

I WILL DO THAT...

7908837174 commented:
Analyzed the original PR: The PR was trying to fix build compatibility issues by downgrading two go-openapi packages:
github.com/go-openapi/runtime from v0.28.0 to v0.27.2
github.com/go-openapi/validate from v0.24.0 to v0.23.0
Identified the core issue: The PR had massive merge conflicts (149 commits, 955 files changed) because the branch was significantly behind master.
Created a clean fix: Instead of resolving thousands of merge conflicts, I:
Created a new branch fix-go-openapi-clean based on the latest master
Applied only the essential go-openapi package downgrades
Updated both go.mod and go.sum files with the correct versions and hashes
Added explanatory comments as requested in the original PR review
Preserved the original fix: The changes include:
go.mod: Downgraded the two packages with explanatory comments
go.sum: Updated with the correct hashes for the downgraded versions
Commit message: Maintained the same detailed explanation as the original
Pushed the clean fix: Force-pushed the clean solution to the original PR branch, replacing the conflicted version with a clean, mergeable version.

7908837174 commented:
Sure!

On Wed, 23 Jul, 2025, 18:42 Naman Verma, ***@***.***> wrote:

> *Namanv0509* left a comment (meshery/meshery#15253)
> <https://github.com/meshery/meshery/pull/15253#issuecomment-3108246363>
>
> @7908837174 <https://github.com/7908837174>
> Thank you for your contributions! Let's discuss this today, during Meshery
> Development call at 9:00 AM CT. Add it as an agenda item to the 📝 meeting
> minutes <https://meet.meshery.io/dev-minutes>, if you would.
> https://meet.meshery.io/dev
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/meshery/meshery/pull/15253#issuecomment-3108246363>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BLR2IIOCTBHPUH45Q44HKDD3J6C3TAVCNFSM6AAAAACAOALSNSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTCMBYGI2DMMZWGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

leecalcote commented:
> Analyzed the original PR: The PR was trying to fix build compatibility issues by downgrading two go-openapi packages: github.com/go-openapi/runtime from v0.28.0 to v0.27.2 github.com/go-openapi/validate from v0.24.0 to v0.23.0 Identified the core issue: The PR had massive merge conflicts (149 commits, 955 files changed) because the branch was significantly behind master. Created a clean fix: Instead of resolving thousands of merge conflicts, I: Created a new branch fix-go-openapi-clean based on the latest master Applied only the essential go-openapi package downgrades Updated both go.mod and go.sum files with the correct versions and hashes Added explanatory comments as requested in the original PR review Preserved the original fix: The changes include: go.mod: Downgraded the two packages with explanatory comments go.sum: Updated with the correct hashes for the downgraded versions Commit message: Maintained the same detailed explanation as the original Pushed the clean fix: Force-pushed the clean solution to the original PR branch, replacing the conflicted version with a clean, mergeable version.

When you copy/paste AI slop, it only serves to extend the time it takes to review PRs.
